src/base/bittorrent/filesearcher.h

```diff
@@ -29,26 +29,23 @@
 #pragma once
 
 #include <QObject>
+#include <QPromise>
 
 #include "base/path.h"
 
-namespace BitTorrent
+struct FileSearchResult
 {
-    class TorrentID;
-}
+    Path savePath;
+    PathList fileNames;
+};
 
 class FileSearcher final : public QObject
 {
-    Q_OBJECT
```

Comment:
why remove it?

Reply:
`Q_OBJECT` macro is mandatory for any object that implements signals, slots or properties. It can be omitted otherwise. But latest Qt docs "strongly recommend the use of this macro in all subclasses of `QObject` regardless of whether or not they actually use signals, slots and properties, since failure to do so may lead certain functions to exhibit strange behavior" so I returned it back.